src/app/clusters/testing/ClusterTester.h

```diff
@@ -18,110 +18,152 @@
 #include <app/AttributeValueEncoder.h>
 #include <app/CommandHandler.h>
 #include <app/ConcreteAttributePath.h>
+#include <app/ConcreteClusterPath.h>
+#include <app/ConcreteEventPath.h>
+#include <app/data-model-provider/ActionReturnStatus.h>
 #include <app/data-model-provider/tests/ReadTesting.h>
 #include <app/data-model-provider/tests/WriteTesting.h>
 #include <app/server-cluster/ServerClusterInterface.h>
+#include <app/server-cluster/testing/TestServerClusterContext.h>
+#include <lib/core/CHIPError.h>
 #include <lib/core/TLVReader.h>
+#include <lib/support/ReadOnlyBuffer.h>
+
 #include <memory>
+#include <vector>
 
 namespace chip {
 namespace Test {
 
 // Helper class for testing clusters.
 //
-// This class ensures that data read by attribute is referencing valid memory for one
-// read request (e.g. for case where the underlying read references a list or string that
-// points to TLV data).
-//
-// Calling `ReadAttribute` again may invalidate the result of a previous `ReadAttribute`.
+// This class ensures that data read by attribute is referencing valid memory for all
+// read requests until the ClusterTester object goes out of scope. (for the case where the underlying read references a list or
+// string that points to TLV data).
 //
 // Read/Write of all attribute types should work, but make sure to use ::Type for encoding
 // and ::DecodableType for decoding structure types.
 //
 // Example of usage:
 //
-// FixedLabelCluster fixedLabel(kRootEndpointId);
+// ExampleCluster cluster(someEndpointId);
 //
-// ClusterTester tester(fixedLabel);
-// uint32_t features{};
+// // Possibly steps to setup the cluster
+//
+// ClusterTester tester(cluster);
+// app::Clusters::ExampleCluster::Attributes::FeatureMap::TypeInfo::DecodableType features;
 // ASSERT_EQ(tester.ReadAttribute(FeatureMap::Id, features), CHIP_NO_ERROR);
 //
-// DataModel::DecodableList<Structs::LabelStruct::DecodableType> labelList;
-// ASSERT_EQ(tester.ReadAttribute(LabelList::Id, labelList), CHIP_NO_ERROR);
-// auto it = labelList.begin();
+// app::Clusters::ExampleCluster::Attributes::ExampleListAttribute::TypeInfo::DecodableType list;
+// ASSERT_EQ(tester.ReadAttribute(LabelList::Id, list), CHIP_NO_ERROR);
+// auto it = list.begin();
 // while (it.Next())
 // {
 //     ASSERT_GT(it.GetValue().label.size(), 0u);
 // }
+//
 class ClusterTester
 {
 public:
-    ClusterTester(app::ServerClusterInterface & cluster) : mCluster(cluster) {}
+    // Sets the cluster to be used for Read/Write/Invoke operations.
+    ClusterTester(app::ServerClusterInterface & cluster) : mCluster(&cluster) {}
```

Comment:
why did we switch this to a pointer instead of a reference? It seems reference would work (since we removed the ability to `set`

Comment seems off - this is a constructor, not a setter.